### PR TITLE
Pass synchronizer id on the context of the call to synchronizer

### PR DESCRIPTION
**Pull request by sawagh**

### CODE REVIEW

File: internal/util/synccontextid.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	values := getSynchronizerContextIDFromMetadata(md)
+	if len(values) == 1 {
+		return values[0]
+	}
+
+	return ""
+}
+
+func getSynchronizerContextIDFromMetadata(md metadata.MD) []string {
+	values := md.Get(metadataNameSynchronizerContextID)
```
Comment (yfei1) [inline internal/util/synccontextid.go]: Is the following code the same thing as `return md.Get(metadataNameSynchronizerContextID)`?

File: internal/util/synccontextid.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+)
+
+// AppendSynchronizerContextID adds the synchronizer context id to a request context metadata.
+func AppendSynchronizerContextID(ctx context.Context, id string) (context.Context, error) {
+	md, ok := metadata.FromOutgoingContext(ctx)
+	if ok {
+		values := getSynchronizerContextIDFromMetadata(md)
+		if len(values) == 1 {
+			if values[0] != id {
+				return ctx, errors.New("request already has a synchronizer context id")
```
Comment (yfei1) [inline internal/util/synccontextid.go]: maybe `request already has a different synchronizer context id`

Comment (sawagh) [reply internal/util/synccontextid.go]: done.

File: internal/util/synccontextid.go (hunk 1 above)
Comment (sawagh) [reply internal/util/synccontextid.go]: That would get nil in some cases - might be more readable to return empty slice for len on it to be

**Pull request merged**